src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorFeatureDetector.cs

```diff
@@ -159,11 +187,11 @@ private protected virtual bool EnvironmentFeatureEnabled()
         }
 
         // Private protected virtual for testing
-        private protected virtual bool IsFeatureFlagEnabled()
+        private protected virtual bool IsFeatureFlagEnabledCached()
         {
-            if (!_featureFlagEnabled.HasValue)
-            {
-                _featureFlagEnabled = _featureFlags.IsFeatureEnabled(RazorLSPEditorFeatureFlag, defaultValue: false);
+            if (!_featureFlagEnabled.HasValue)
+            {
+                _featureFlagEnabled = _featureFlags.IsFeatureEnabled(RazorLSPEditorFeatureFlag, defaultValue: false);
```

Comment:
The actual fix for the IsFeatureEnabled performance is actually in, so caching shouldn't really be necessary any more.

Reply:
Here's the PR: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/260509

Reply:
Is there any harm in keeping it in? We call this a *lot*, it's one lesser check in the hot path.

Reply:
The only concern I have is now that we cache this (and the cache that I added for another feature flag check earlier), that the razor LS requires a restart after setting the feature flag.

Reply:
I don't for-see users switching flags too frequently, and the benefits they get in improved user experience are significant. 

From your change earlier, we already specify users must restart the language server after setting the feature flag: https://github.com/dotnet/aspnetcore-tooling/pull/2037/files#diff-c427b3bec8fcf6b87d83c4641cbc33d3R17